app/modules/asset/service.server.ts

```diff
@@ -2420,3 +2449,304 @@ export async function createKitChangeNote({
+}) {
+  try {
+    /**
+     * In order to make notes for the assets we have to make this query to get info about assets
+     */
+    const [assets, user] = await Promise.all([
+      db.asset.findMany({
+        where: { id: { in: assetIds } },
+        select: {
+          id: true,
+          title: true,
+          custody: {
+            select: { id: true, custodian: { include: { user: true } } },
+          },
+        },
+      }),
+      getUserByID(userId),
+    ]);
+
+    /**
+     * updateMany does not allow to update nested relationship rows
+     * so we have to make two queries to bulk check-in assets
+     * 1. Delete all custodies for all assets
+     * 2. Update status of all assets to AVAILABLE
+     */
+    await db.$transaction(async (tx) => {
+      /** Deleting custodies over assets */
+      await tx.custody.deleteMany({
+        where: {
+          id: {
+            in: assets.map((asset) => {
+              /** This case should not happen but in case */
+              if (!asset.custody) {
+                throw new Error("Could not find custody over asset.");
+              }
+
+              return asset.custody.id;
+            }),
+          },
+        },
+      });
+
+      /** Updating status of assets to AVAILABLE */
+      await tx.asset.updateMany({
+        where: { id: { in: assets.map((asset) => asset.id) } },
+        data: { status: AssetStatus.AVAILABLE },
+      });
+
+      /** Creating notes for the assets */
+      await tx.note.createMany({
+        data: assets.map((asset) => ({
+          content: `**${user.firstName?.trim()} ${
+            user.lastName
+          }** has released **${resolveTeamMemberName(
+            asset.custody!.custodian
+          )}'s** custody over **${asset.title?.trim()}**`,
+          type: "UPDATE",
+          userId,
+          assetId: asset.id,
+        })),
+      });
+    });
+
+    return true;
+  } catch (cause) {
+    throw new ShelfError({
+      cause,
+      message: "Something went wrong while bulk checking in assets.",
+      additionalData: { assetIds, userId },
+      label,
+    });
+  }
+}
+
+export async function bulkUpdateAssetLocation({
+  userId,
+  assetIds,
+  organizationId,
+  newLocationId,
+  currentLocationId,
+}: {
+  userId: User["id"];
+  assetIds: Asset["id"][];
+  organizationId: Asset["organizationId"];
+  newLocationId?: Location["id"] | null;
+  currentLocationId?: Location["id"];
+}) {
+  try {
+    /** We have to create notes for all the assets so we have make this query */
+    const [assets, user] = await Promise.all([
+      db.asset.findMany({
+        where: { id: { in: assetIds }, organizationId },
+        select: { id: true, title: true },
+      }),
+      getUserByID(userId),
+    ]);
+
+    const newLocation = newLocationId
+      ? await db.location.findFirst({
+          where: { id: newLocationId, organizationId },
+        })
+      : null;
+
+    const currentLocation = currentLocationId
```

Comment:
Hmm we have to re-consider this. I think we have to query the locations of all assets and do this check for each asset. As there could be multiple assets that are being assigned to Location A but each of those assets have different locations. That should be possible. So having said that I think we should consider something like this:

1. The new location for all assets is the same
2. The currentLocation of all assets could be different

So considering this i think we should:
- [ ] include the current location data needed for the logic, in the assets query
- [ ] Iterate over the assets and make the old/new location comparison for each asset so we know what note we have to create for it